cmake/Gems.cmake

```diff
@@ -252,3 +252,77 @@ function(ly_enable_gems_delayed)
         endforeach()
     endforeach()
 endfunction()
+
+#! ly_is_gem_available: Check if a gem is available in the current project configuration.
+# \arg:NAME name of the gem to check for availability.
+# \arg:RETURN_VARIABLE name of the variable in which to set the returned value.
+# \arg:VALUE_IF_FOUND value to set in the RETURN_VARIABLE when the gem is available. Defaults to TRUE.
+# \arg:VALUE_IF_NOT_FOUND value to set in the RETURN_VARIABLE when the gem is not available. Defaults to FALSE.
+# \arg:VARIANT name of the project variant if which to check for availability. Defaults to 'Clients'.
+function(ly_is_gem_available)
+    set(options)
+    set(oneValueArgs NAME RETURN_VARIABLE VALUE_IF_FOUND VALUE_IF_NOT_FOUND VARIANT)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(ly_is_gem_available "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if (NOT ly_is_gem_available_NAME)
+        message(FATAL_ERROR "Invalid call to 'ly_is_gem_available'. Specify the name of the gem with the 'NAME' argument.")
+    endif()
+
+    if (NOT ly_is_gem_available_VARIANT)
+        set(ly_is_gem_available_VARIANT "Clients")
+    endif()
+
+    if (NOT ly_is_gem_available_VALUE_IF_FOUND)
+        set(ly_is_gem_available_VALUE_IF_FOUND TRUE)
+    endif()
+
+    if (NOT ly_is_gem_available_VALUE_IF_NOT_FOUND)
+        set(ly_is_gem_available_VALUE_IF_NOT_FOUND FALSE)
+    endif()
+
+    if (NOT ly_is_gem_available_RETURN_VARIABLE)
+        message(WARNING "Call to 'ly_is_gem_available' without specifying the RETURN_VARIABLE argument. The call was ignored.")
+        return()
+    endif()
+
+    get_property(ly_delayed_enable_gems GLOBAL PROPERTY LY_DELAYED_ENABLE_GEMS)
+    foreach(project_target_variant ${ly_delayed_enable_gems})
```

Comment:
The output boolean should be initialized to `FALSE` before the foreach loop over the delayed gem targets.